app/components/fade.js

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {Animated} from 'react-native';
+
+export const FADE_DURATION = 300;
+
+export default class Fade extends Component {
+    static propTypes = {
+        children: PropTypes.node.isRequired,
+        style: PropTypes.object,
+        visible: PropTypes.bool.isRequired,
+        disabled: PropTypes.bool.isRequired,
+    };
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            fadeAnim: new Animated.Value(this.props.visible ? 1 : 0),
+        };
+    }
+
+    shouldComponentUpdate(nextProps) {
+        if (this.props.visible !== nextProps.visible || this.props.disabled !== nextProps.disabled) {
```

Comment:
I doubt they'll ever change, but shouldn't this also update if `children` or `style` change as well? Perhaps we could use a `PureComponent` for this instead?

Reply:
Good catch! Will do.